massdriver-application/aws_identity.tf

```diff
@@ -6,9 +6,9 @@ locals {
   }
 
   aws_service                      = local.aws_service_map[var.service]
-  aws_eks_oidc_short               = local.is_aws && local.is_kubernetes ? replace(var.kubernetes.cluster_artifact.data.infrastructure.oidc_issuer_url, "https://", "") : null
-  aws_eks_oidc_federated_principal = local.is_aws && local.is_kubernetes ? "arn:aws:iam::${data.mdxc_cloud.current.id}:oidc-provider/${local.aws_eks_oidc_short}" : null
-  aws_eks_assume_role_conditional  = local.is_aws && local.is_kubernetes ? "system:serviceaccount:${var.kubernetes.namespace}:${var.name}" : null
+  aws_eks_oidc_short               = local.is_aws && local.is_kubernetes ? replace(var.kubernetes.cluster_artifact.data.infrastructure.oidc_issuer_url, "https://", "") : ""
+  aws_eks_oidc_federated_principal = local.is_aws && local.is_kubernetes ? "arn:aws:iam::${data.mdxc_cloud.current.id}:oidc-provider/${local.aws_eks_oidc_short}" : ""
+  aws_eks_assume_role_conditional  = local.is_aws && local.is_kubernetes ? "system:serviceaccount:${var.kubernetes.namespace}:${var.name}" : ""
```

Comment:
Why are we changing the types from null to `""`?

Reply:
Good question! I meant to call this out. If these are null, the locals in`massdriver-application/aws_identity.tf` produce an error.  

```
│ Error: Invalid template interpolation value
│
│   on .terraform/modules/application/massdriver-application/aws_identity.tf line 44, in locals:
│   36:   aws_federated_principal_assume_role = <<EOF
│   37: {
│   38:   "Version": "2012-10-17",
│   39:   "Statement": [
│   40:     {
│   41:       "Effect": "Allow",
│   42:       "Action": "sts:AssumeRoleWithWebIdentity",
│   43:       "Principal": {
│   44:         "Federated": "${local.aws_eks_oidc_federated_principal}"
│   45:       },
│   46:       "Condition": {
│   47:         "StringEquals": {
│   48:           "${local.aws_eks_oidc_short}:sub": "${local.aws_eks_assume_role_conditional}"
│   49:         }
│   50:       }
│   51:     }
│   52:   ]
│   53: }
│   54: EOF
│     ├────────────────
│     │ local.aws_eks_oidc_federated_principal is null
│
│ The expression result is null. Cannot include a null value in a string template.
```

Reply:
Can we put ternaries on the two aws locals instead so the pattern is consistent. Id hate for someone to get a policy that looks right, until they notice a field is empty and it was supposed to be _no policy_

Reply:
Yeah no problem, I can move the ternary into this file.

Reply:
You'll need the ternary up there, im saying:

`aws_federated_principal_assume_role = local.is_aws ? the value : null`

Reply:
Yeah I got you 💪 . I went that route and just updated it to use `is_aws` instead of checking null for the other locals.